src/messaging/messaging.ios.ts

```diff
@@ -387,33 +392,35 @@ function _registerForRemoteNotifications() {
       }
     });
 
-    _userNotificationCenterDelegate = UNUserNotificationCenterDelegateImpl.new().initWithCallback((unnotification, actionIdentifier?, inputText?) => {
-      // if the app is in the foreground then this method will receive the notification
-      // if the app is in the background, and user has responded to interactive notification, then this method will receive the notification
-      // if the app is in the background, and user views a notification, applicationDidReceiveRemoteNotificationFetchCompletionHandler will receive it
-      const userInfo = unnotification.request.content.userInfo;
-      const userInfoJSON = firebaseUtils.toJsObject(userInfo);
-      updateUserInfo(userInfoJSON);
-
-      if (actionIdentifier) {
-        _pendingActionTakenNotifications.push({
-          actionIdentifier,
-          userInfoJSON,
-          inputText
-        });
-        if (_notificationActionTakenCallback) {
-          _processPendingActionTakenNotifications();
+    if (_displayNotifications) {
+      _userNotificationCenterDelegate = UNUserNotificationCenterDelegateImpl.new().initWithCallback((unnotification, actionIdentifier?, inputText?) => {
+        // if the app is in the foreground then this method will receive the notification
+        // if the app is in the background, and user has responded to interactive notification, then this method will receive the notification
+        // if the app is in the background, and user views a notification, applicationDidReceiveRemoteNotificationFetchCompletionHandler will receive it
+        const userInfo = unnotification.request.content.userInfo;
+        const userInfoJSON = firebaseUtils.toJsObject(userInfo);
+        updateUserInfo(userInfoJSON);
+
+        if (actionIdentifier) {
+          _pendingActionTakenNotifications.push({
+            actionIdentifier,
+            userInfoJSON,
+            inputText
+          });
+          if (_notificationActionTakenCallback) {
+            _processPendingActionTakenNotifications();
+          }
         }
-      }
 
-      userInfoJSON.foreground = true;
-      _pendingNotifications.push(userInfoJSON);
-      if (_receivedNotificationCallback) {
-        _processPendingNotifications();
-      }
-    });
+        userInfoJSON.foreground = true;
+        _pendingNotifications.push(userInfoJSON);
+        if (_receivedNotificationCallback) {
+          _processPendingNotifications();
+        }
+      });
 
-    curNotCenter.delegate = _userNotificationCenterDelegate;
+      curNotCenter.delegate = _userNotificationCenterDelegate;
```

Comment:
The change here is basically to avoid this, as multiple plugins might be doing that and the `NotificationCenter` is a singleton, so only one plugin can set its custom `.delegate` at a time.